content/assignments/techjobs-console/_index.md

```diff
@@ -0,0 +1,477 @@
+re-prompts the user.
+
+The local variable `choiceKeys` is used to easily enumerate the
+`choices` `Dictionary`. In other words, it gives us a simple way to
+provide an ordering to `choices`, which doesn't have an ordering of
+its own.
+
+## The `JobData` Class
+
+The `JobData` class is responsible for importing the data from the CSV
+file and parsing it into a C#-friendly format, that is, into
+`Dictionary` and `List` form. Look toward the bottom of the class
+and you will see a method named `LoadData()`, which does just what it
+advertises. After parsing the file data, it stores the data in the
+private property `AllJobs` which is of type
+`List<Dictionary<string, string>>`.
+
+{{% notice blue "Note" "rocket"%}}
+
+   We haven't covered static properties and methods in-depth yet. For this
+   assignment, know simply that they allow us to use properties and methods
+   of a class without creating an object from that class. For example, we
+   can call `JobData.FindAll()` from the `TechJob` class.
+
+   If you want to create a new method in `JobData`, or add a property, be
+   sure to declare it as `static`.
+
+{{% /notice %}}
+
+Let's look more closely at the data type of `AllJobs`. It purports to
+be a `List` that stores `Dictionary` objects which have
+`string` keys and `string` values. If we were to represent some of
+this data visually, using `[]` for a `List` and `{}` for a collection of
+key/value pairs (i.e., a `Dictionary`), it would look like this:
+
+```csharp {linenos=table}
+   [
+       {
+           "name": "Junior Data Analyst",
+           "employer": "Lockerdome",
+           "location": "Saint Louis",
+           "position type": "Data Scientist / Business Intelligence",
+           "core competency": "Statistical Analysis"
+       },
+       {
+           "name": "Junior Web Developer",
+           "employer": "Cozy",
+           "location": "Portland",
+           "position type": "Web - Back End",
+           "core competency": "Ruby"
+       },
+       ...
+   ]
+```
+
+If you look at the `LoadData()` method you'll see a lot of unfamiliar code.
+Blake wrote this essential piece of code for you, and while you won't have to
+modify it, it will be useful to have an idea of how it works. Read
+through the code until you feel like you can describe its functionality
+at a basic level.
+
+There are three more methods in `JobData`, each of which is public
+(and `static`, per our earlier note): 
+- `FindAll()`
+- `FindAll(string)`
+- `FindByColumnAndValue(string, string)`. 
+
+Note that there are two methods named `FindAll()`, but this is allowed in
+C# via a feature called **overloading**. Overloading happens when
+multiple methods have the same name, but they each have different input
+parameters (also called argument lists). Read more about
+[overloading](https://www.geeksforgeeks.org/c-sharp-method-overloading/).
+
+Here are some questions to ask yourself while reading this code:
```

Comment:
```suggestion
Questions to ask yourself while reading thru this code:
```

Reply:
Thru is only to be used in really informal writing or when talking about drive-thru restaurants and ATMS.

Reply:
While I appreciate the reply based on Grammarly's interpretation. Thru is also used in place of through in less formal places.